### ISSUE TITLE
After upgrading to Flutter 3.16 and XCode 15.0.1 I get Error vailability_version_check.cc(39)] Check failed on iOS less than 17

### ISSUE THREAD

**Issue by JacoFourie** [flutter/flutter#138711]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

I upgraded my Flutter to version 3.16 and XCode 15.0.1

### Expected results

The app should work after the upgrade

### Actual results

Installing and launching...
[FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.

### Code sample

<details><summary>Code sample</summary>

```dart
Create a blank Flutter project.
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
aunching lib/main.dart on Developmet iphone 6 in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: C...........
Running pod install...
Running Xcode build...
Xcode build done.                                           39,6s
Installing and launching...
[FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.1.1 23B81 darwin-arm64, locale en-ZA)
    • Flutter version 3.16.0 on channel stable at /Users/jacofourie/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (5 days ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/jacofourie/Library/Android/sdk
    • Platform android-34, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.14.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • Jaco’s iPhone 14 (mobile)    • 00008110-00110C9E1105401E                • ios            • iOS 17.1.1 21B91
    • Developmet iphone 6 (mobile) • 4781749bcc830a01bf3326c466d1d86cea0bc844 • ios            • iOS 12.5.7 16H81
    • macOS (desktop)              • macos                                    • darwin-arm64   • macOS 14.1.1 23B81 darwin-arm64
    • Chrome (web)                 • chrome                                   • web-javascript • Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by josh2112** [comment]
Same for me. Runs fine in the iPhone simulator though.

My development iPod is quite old and running iOS version 12.5.5 (I notice you are running iOS 12.5.7, @JacoFourie ). I suspect the minimum required iOS version has been bumped up and we're just out of luck :-(

**Comment by mossmana** [comment]
@vashworth Is there a version compatibility matrix we can link to?

**Comment by JacoFourie** [comment]
> Same for me. Runs fine in the iPhone simulator though.
> 
> The app appears to get installed (the icon appears in the launcher), but when I tap it it shows the launch image then crashes back to the launcher, printing the following message in the Flutter debug console:
> 
> `[FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.`
> 
> My development iPod is quite old and running iOS version 12.5.5 (I notice you are running iOS 12.5.7, @JacoFourie ). I suspect the minimum required iOS version has been bumped up and we're just out of luck :-(

https://docs.flutter.dev/reference/supported-platforms

This link states 11 - 15 is best effort, so it should still work. It was not in the breaking changes list. The iPhone 6 still runs my app fine.

**Comment by juliansteenbakker** [comment]
Xcode 15 does not support iOS 11 (see https://github.com/fluttercommunity/plus_plugins/pull/2169).

I had the same issue, but fixed it by bumping the deployment target to iOS 12 in the podfile, and cleaning and rebuilding the project. (https://stackoverflow.com/a/64075795/13031778 but use iOS 12)

**Comment by zanderso** [comment]
I suspect that https://github.com/flutter/flutter/issues/138711#issuecomment-1819758879 is the answer here.

The availability check implementation added in https://github.com/flutter/engine/pull/44711 was copied from clang's implementation from its iOS runtime library prior to an ABI breaking change to clang, which allowed Flutter to roll clang forward across the ABI break. Which is to say, the failures on iOS 12 in this issue and https://github.com/flutter/flutter/issues/135317 would be happening even without https://github.com/flutter/engine/pull/44711 and leaving clang pinned to the old version.

cc @chinmaygarde @leighajarett

**Comment by JacoFourie** [comment]
> Xcode 15 does not support iOS 11 (see [fluttercommunity/plus_plugins#2169](https://github.com/fluttercommunity/plus_plugins/pull/2169)).
> 
> I had the same issue, but fixed it by bumping the deployment target to iOS 12 in the podfile, and cleaning and rebuilding the project. (https://stackoverflow.com/a/64075795/13031778 but use iOS 12)

Well this is obviously not the case as my device is iOS 12

**Comment by CyJaySong** [comment]
3.16 also has the same problem

**Comment by nakhunchumpolsathien** [comment]
3.16 also has the same problem

**Comment by juliansteenbakker** [comment]
It seems that my previous answer is incorrect. All of my other projects are failing on iOS 12 with Flutter 3.16 on this error, also on the Beta channel.

**Comment by diogofos** [comment]
I had the same problem. Using the latest version 3.16 of Flutter. The AvailabilityVersionCheck error occurs. I went back to version 3.10.6 and the error stopped occurring.

**Comment by zanderso** [comment]
Does this issue reproduce only with Xcode 15, or does it also happen with earlier versions of Xcode?

Also, folks should be aware of Flutter's issue hygiene policies, especially around comments: https://github.com/flutter/flutter/wiki/Issue-hygiene#comments. Please do not make "me too" comments that don't contribute new information---use the thumbs-up reaction on the original issue report, instead.

**Comment by egops** [comment]
Why was this issue assigned a P1 priority? It should be P0, shouldn't it?

**Comment by JacoFourie** [comment]
What is the status of this issue? It is a major one. You can not upgrade to version 3.16 as all your iOS version 12 and lower breaks. I only have an iOS 12 device.

**Comment by felipecastrosales** [comment]
Also happening here. 
Every device with iOS 12.x is currently crashing.
Downgrading to 3.13.9 works as expected. I also think this issue should be P0.

**Comment by szuwest** [comment]
Have the same problem here.
Iphone6 ios 12.5.7

**Comment by daomydong1995** [comment]
i was changed channel master but i still get  EXC_BAD_ACCESS (code=1, address=0x1170000) on ios

**Comment by juliansteenbakker** [comment]
I can confirm that my app is working on `Channel master, 3.18.0-7.0.pre.88, on macOS 14.1.2 23B92 darwin-x64, locale nl-NL`. The app however does crash after clicking on the home button `(Thread 18: EXC_BAD_ACCESS (code=1, address=0xe04000))`, but i don't think it is related to this issue.

**Comment by dickverweij** [comment]
What is de ETA for channel stable for this fix?

**Comment by zanderso** [comment]
@juliansteenbakker @daomydong1995 I suspect that you might be hitting a different bug. Would you mind filing a new issue with full details?

**Comment by chinmaygarde** [comment]
FWIW, I am not able to reproduce the home button crash in the sample app.

**Comment by gokhanikisivri** [comment]
@daomydong1995 @juliansteenbakker  I was encountering a similar error. When I updated Xcode to version 15.1, the issue was resolved.

**Comment by zanderso** [comment]
The fix mentioned above was included in the 3.16.4 stable hotfix release.

**Comment by zanderso** [comment]
Actually, apologies. The release announcement was mistaken. The change mentioned above will go out in the hotfix release next week.
